### ISSUE TITLE
Behaviour with repeat when headers set in response seems to execute an extra time.

### ISSUE THREAD

**Issue by loneparadox** [bbyars/mountebank#158]
Possibly related to #130 
#### Expected behaviour

When creating the following imposter:

```
{
  "port": 4545,
  "protocol": "http",
  "stubs": [
    {
      "responses": [
        {
          "is": {
            "body": "This will only repeat 2 times",
            "headers": {
              "Content-Type": "application/xml"
            }
          },
          "_behaviors": {
            "repeat": 2
          }
        },
        {
          "is": {
            "body": "Then this will start responding"
          }
        }
      ]
    }
  ]
}
```

We expect the first response to be called twice. then the second response to be returned the third time.
...
#### Actual behaviour

The first 3 calls responds with `"This will only repeat 2 times"` then the 4th call responds with `"Then this will start responding"`.
...
#### Steps to reproduce

Create the above imposter and call it 4 times.
Note the only difference between the above request and the request depicted on http://www.mbtest.org/docs/api/behaviors is the addition of the headers in the response.

...
#### Software versions used

```
OS         : Ubuntu 14.04
mountebank : 1.6.0
node.js    : 0.10.26
Installation method : npm
```
#### Log contents in mb.log when running mb --loglevel debug

```
debug: [mb:2525] 127.0.0.1:49788 => {"port":4545,"protocol":"http","stubs":[{"responses":[{"is":{"body":"This will only repeat 2 times","headers":{"Content-Type":"application/xml"}},"_behaviors":{"repeat":2}},{"is":{"body":"Then this will start responding"}}]}]}
info: [http:4545] Open for business...
debug: [http:4545] 127.0.0.1:41320 ESTABLISHED
info: [http:4545] 127.0.0.1:41320 => GET /customers/123
debug: [http:4545] 127.0.0.1:41320 => {"requestFrom":"127.0.0.1:41320","method":"GET","path":"/customers/123","query":{},"headers":{"Host":"127.0.0.1:4545","Connection":"keep-alive","CSP":"active","Cache-Control":"no-cache","User-Agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.130 Safari/537.36","Postman-Token":"f49a57dd-7241-f61a-fa9a-0de07b329f4d","Accept":"*/*","Accept-Encoding":"gzip, deflate, sdch","Accept-Language":"en-GB,en-US;q=0.8,en;q=0.6"},"body":""}
debug: [http:4545] 127.0.0.1:41320 using predicate match: {}
debug: [http:4545] 127.0.0.1:41320 generating response from {"is":{"body":"This will only repeat 2 times","headers":{"Content-Type":"application/xml"}},"_behaviors":{"repeat":2}}
debug: [http:4545] 127.0.0.1:41320 using stub response behavior {"repeat":2}
debug: [http:4545] 127.0.0.1:41320 <= {"statusCode":200,"headers":{"Content-Type":"application/xml","Connection":"close"},"body":"This will only repeat 2 times","_mode":"text"}
debug: [http:4545] 127.0.0.1:41320 CLOSED
debug: [http:4545] 127.0.0.1:41321 ESTABLISHED
info: [http:4545] 127.0.0.1:41321 => GET /customers/123
debug: [http:4545] 127.0.0.1:41321 => {"requestFrom":"127.0.0.1:41321","method":"GET","path":"/customers/123","query":{},"headers":{"Host":"127.0.0.1:4545","Connection":"keep-alive","CSP":"active","Cache-Control":"no-cache","User-Agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.130 Safari/537.36","Postman-Token":"3bbc2827-fb31-6019-65d9-33aee4c95295","Accept":"*/*","Accept-Encoding":"gzip, deflate, sdch","Accept-Language":"en-GB,en-US;q=0.8,en;q=0.6"},"body":""}
debug: [http:4545] 127.0.0.1:41321 using predicate match: {}
debug: [http:4545] 127.0.0.1:41321 generating response from {"is":{"body":"This will only repeat 2 times","headers":{"Content-Type":"application/xml","Connection":"close"}},"_behaviors":{"repeat":2}}
debug: [http:4545] 127.0.0.1:41321 using stub response behavior {"repeat":2}
debug: [http:4545] 127.0.0.1:41321 <= {"statusCode":200,"headers":{"Content-Type":"application/xml","Connection":"close"},"body":"This will only repeat 2 times","_mode":"text"}
debug: [http:4545] 127.0.0.1:41321 CLOSED
debug: [http:4545] 127.0.0.1:41322 ESTABLISHED
info: [http:4545] 127.0.0.1:41322 => GET /customers/123
debug: [http:4545] 127.0.0.1:41322 => {"requestFrom":"127.0.0.1:41322","method":"GET","path":"/customers/123","query":{},"headers":{"Host":"127.0.0.1:4545","Connection":"keep-alive","CSP":"active","Cache-Control":"no-cache","User-Agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.130 Safari/537.36","Postman-Token":"68a931b6-3cff-fd3f-1bd9-92ea24353531","Accept":"*/*","Accept-Encoding":"gzip, deflate, sdch","Accept-Language":"en-GB,en-US;q=0.8,en;q=0.6"},"body":""}
debug: [http:4545] 127.0.0.1:41322 using predicate match: {}
debug: [http:4545] 127.0.0.1:41322 generating response from {"is":{"body":"This will only repeat 2 times","headers":{"Content-Type":"application/xml","Connection":"close"}},"_behaviors":{"repeat":2}}
debug: [http:4545] 127.0.0.1:41322 using stub response behavior {"repeat":2}
debug: [http:4545] 127.0.0.1:41322 <= {"statusCode":200,"headers":{"Content-Type":"application/xml","Connection":"close"},"body":"This will only repeat 2 times","_mode":"text"}
debug: [http:4545] 127.0.0.1:41322 CLOSED
debug: [http:4545] 127.0.0.1:41323 ESTABLISHED
info: [http:4545] 127.0.0.1:41323 => GET /customers/123
debug: [http:4545] 127.0.0.1:41323 => {"requestFrom":"127.0.0.1:41323","method":"GET","path":"/customers/123","query":{},"headers":{"Host":"127.0.0.1:4545","Connection":"keep-alive","CSP":"active","Cache-Control":"no-cache","User-Agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.130 Safari/537.36","Postman-Token":"4d3a3c2f-cb71-611c-9a00-46b79700f574","Accept":"*/*","Accept-Encoding":"gzip, deflate, sdch","Accept-Language":"en-GB,en-US;q=0.8,en;q=0.6"},"body":""}
debug: [http:4545] 127.0.0.1:41323 using predicate match: {}
debug: [http:4545] 127.0.0.1:41323 generating response from {"is":{"body":"Then this will start responding"}}
debug: [http:4545] 127.0.0.1:41323 <= {"statusCode":200,"headers":{"Connection":"close"},"body":"Then this will start responding","_mode":"text"}
debug: [http:4545] 127.0.0.1:41323 CLOSED
```

**Comment by njenan** [comment]
So it looks like the immediate problem is that a `Connection: close` header is getting added to the stub somehow.  If you retrieve the imposter after the second request you will see the following:
```javascript
{
  "port": 4545,
  "protocol": "http",
  "stubs": [
    {
      "responses": [
        {
          "is": {
            "body": "This will only repeat 2 times",
            "headers": {
              "Content-Type": "application/xml",
              "Connection": "close" <----------------- Extra field
            }
          },
          "_behaviors": {
            "repeat": 2
          }
        },
        {
          "is": {
            "body": "Then this will start responding"
          }
        }
      ]
    }
  ]
}
```

A workaround for the time being would be to add the extra header to your imposter when you post it, Mountebank appears to function as expected when that is done.

I still don't know why this header is being added though, but I will keep looking,

**Comment by bbyars** [comment]
That gets added in baseHttpServer (lines 58-63). I actually am not sure if
it's needed or not. I wrote a precursor to mountebank many years ago called
httpmock, and when testing a Java binding for it I remember having to add
in the connection close for some concurrent testing. That design decision
was copied into mountebank because I remember it being a tricky issue to
diagnose, but I don't recall the details.

I'll test taking it out over the next few weeks and seeing if some teams
who use it for large scale testing see any ill effects.
-Brandon

**Comment by bbyars** [comment]
Hacked a solution, will revisit for a more general purpose one. Fixed in 0c35cf8
